dirty_cat/_fuzzy_join.py

```diff
@@ -230,49 +195,31 @@ def fuzzy_join(
     distance, neighbors = neigh.kneighbors(left_enc, return_distance=True)
     idx_closest = np.ravel(neighbors)
 
-    norm_distance = 1 - (distance / 2)
-
-    left_array = np.array(left_table_clean)
-    right_array = np.array(right_table_clean)
-    joined = np.append(
-        left_array,
-        np.array(
-            [
-                right_array[idx_closest[idr]]
-                if norm_distance[idr] >= match_score
-                else np.tile(np.nan, (right_array.shape[1],))
-                for idr in left_table_clean.index
-            ]
-        ),
-        axis=1,
-    )
+    left_table_clean["fj_idx"] = idx_closest
+    right_table_clean["fj_idx"] = right_table_clean.index
 
-    cols = list(left_table_clean.columns) + list(right_table_clean.columns)
-    df_joined = pd.DataFrame(joined, columns=cols).replace(r"^\s*$", np.nan, regex=True)
+    norm_distance = np.round(1 - (distance / 2), 6)
```

Comment:
Why do we round?

Reply:
You are right, not sure how this got in.